transport/src/main/java/io/netty/bootstrap/Bootstrap.java

```diff
@@ -240,18 +272,28 @@ public void operationComplete(Future<SocketAddress> future) throws Exception {
     }
 
     private static void doConnect(
-            final SocketAddress remoteAddress, final SocketAddress localAddress, final ChannelPromise connectPromise) {
+            final SocketAddress remoteAddress, final SocketAddressFunction localAddressFunction,
+            final ChannelPromise connectPromise) {
 
         // This method is invoked before channelRegistered() is triggered.  Give user handlers a chance to set up
         // the pipeline in its channelRegistered() implementation.
         final Channel channel = connectPromise.channel();
         channel.eventLoop().execute(new Runnable() {
             @Override
             public void run() {
-                if (localAddress == null) {
+                if (localAddressFunction == null) {
```

Comment:
Isn't this always non-null because of NULL_FUNCTION?

Reply:
@ejona86 I guess the user could still pass in `null` by himself / herself ?

Reply:
Ah, I see it now. It seems like either 1) get rid of NULL_FUNCTION when a SocketAddress is passed or 2) fail or convert null function to NULL_FUNCTION when function is passed.